### ISSUE TITLE
Improve the usability of the editor

### ISSUE THREAD

**Issue by krishnagopinath** [Matterwiki/Matterwiki#117]
The current Matterwiki editor (in the `refactor` branch) uses `draft-js` as its core editor. While this is good, there are wiggles, focus issues and other small issues that all add up to making the experience of using it sub-par. We could switch to a new editor or make the editor we have better. Any thoughts and suggestions would be helpful!

**Comment by spaceraccoon** [comment]
I've been working with tinyMCE on another project - fairly usable but not
the most ideal!

**Comment by krishnagopinath** [comment]
I have seen tinyMCE too. 
* It's not React compatible in the way I'd like it to be (there was just a thin wrapper around the editor, iirc)
* It feels like a closed source product (even though it IS open source) and all that. 

I personally like draft-js, but if we cant get the UX we need out of it, we could consider http://slatejs.org/, which is similar to draft

**Comment by spaceraccoon** [comment]
Agreed on those fronts - slatejs looks like a solid alternative option.

**Comment by nshntarora** [comment]
How about we check out Mobiledoc too? Not that elaborate as Slate. Slate still gets my vote. But a great piece of software http://bustle.github.io/mobiledoc-kit/demo/
Also, the guys at Ghost moved to a mobiledoc based editor in Ghost 1.0, but then again, their use case is just markdown.
What do you think @krishnagopinath and @spaceraccoon ?

I suggest, let's just make a list of all the editors and then do a pros-and-cons list. We have to ship a new editor (with image support) in the next version.

**Comment by krishnagopinath** [comment]
@nshntarora I think I'll start working on this next..

**Comment by krishnagopinath** [comment]
I'm leaning towards using slatejs on this one because 

* it seems to allow a little more customization through "plugins".. This is fantastic, because in the future, things could be easily made into its own package, if we started using lerna or yarn workspaces..

* it just uses the DOM tree to represent the editor state, instead of its own structure (which fb/draft does), which is more intuitive and easy to reason about.. 

* Easy moving around between HTML, MD and more! This is especially useful for us, because we need to support all these modes. Using draft would force us to re-invent the wheel on this one (Im sure it's possible, but we could be working on something else instead of this for now)
